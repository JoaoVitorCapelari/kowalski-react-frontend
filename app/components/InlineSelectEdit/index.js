/**
*
* InlineSelectEdit
*
*/

import PropTypes from 'prop-types';
import React from 'react';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
import InlineEdit from '../InlineEdit';

const renderSelect = (options, { content, onCommit, onChange, onDiscard }) => (
  <Select
    autoFocus
    onBlurResetsInput={false}
    onSelectResetsInput={false}
    clearable={false}
    value={content}
    onChange={({ value }) => { onChange(value); onCommit(); }}
    onBlur={onDiscard}
    options={options}
    searchable
    autosize={false}
    openOnFocus
    placeholder="Select a user..."
  />
);

const valueFormatterForSelect = (options, toBeFormatted) => {
  const correspondingOption = options.find(({ value }) => toBeFormatted === value);
  if (!correspondingOption) {
    return 'Not defined';
  }

  return correspondingOption.label;
};

function InlineSelectEdit(props) {
  return (
    <InlineEdit
      {...props}
      formatter={(value) => valueFormatterForSelect(props.options, value)}
      renderEditComponent={({ ...editProps }) => renderSelect(props.options, editProps)}
    >
    </InlineEdit>
  );
}

InlineSelectEdit.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.any.isRequired,
  })),
};

export default InlineSelectEdit;
