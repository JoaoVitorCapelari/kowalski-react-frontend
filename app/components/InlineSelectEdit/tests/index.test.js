import React from 'react';
import { shallow } from 'enzyme';

import InlineSelectEdit from '../index';

describe('<InlineSelectEdit />', () => {
  it('renders', () => {
    const renderedComponent = shallow(
      <InlineSelectEdit options={[]} onCommit={jest.fn()} />
    );
    expect(renderedComponent.html()).toMatchSnapshot();
  });
});
