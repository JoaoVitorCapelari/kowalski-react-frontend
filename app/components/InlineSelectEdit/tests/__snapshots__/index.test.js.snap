// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineSelectEdit /> renders 1`] = `"<div class=\\"InlineEdit__Container-iwjTUk hrrFYE\\" title=\\"Click for edit\\"><div class=\\"InlineEdit__ContentContainer-kvxWLC cdZpQT\\">Not defined</div><a class=\\"InlineEdit__ButtonsContainer-cWlMqY hnRZzJ\\"><div class=\\"InlineEdit__EditButton-cDVLLs vgvHz\\"><svg fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\" height=\\"18\\" width=\\"18\\" viewBox=\\"0 0 40 40\\" style=\\"vertical-align:middle;\\"><g><path d=\\"m11.1 34.3l2-2-5.2-5.3-2 2v2.4h2.8v2.9h2.4z m11.7-20.7q0-0.5-0.5-0.5-0.2 0-0.4 0.1l-12.1 12.1q-0.1 0.2-0.1 0.4 0 0.5 0.4 0.5 0.3 0 0.4-0.2l12.1-12.1q0.2-0.1 0.2-0.3z m-1.2-4.3l9.3 9.3-18.6 18.5h-9.3v-9.2z m15.2 2.1q0 1.2-0.8 2l-3.7 3.7-9.3-9.2 3.7-3.7q0.8-0.9 2-0.9 1.2 0 2 0.9l5.3 5.2q0.8 0.9 0.8 2z\\"></path></g></svg></div></a></div>"`;
